lib/engine/ability/tile_lay.rb

```diff
@@ -5,12 +5,13 @@
 module Engine
   module Ability
     class TileLay < Base
-      attr_reader :hexes, :tiles, :free
+      attr_reader :hexes, :tiles, :free, :discount
 
-      def setup(hexes:, tiles:, free: false)
+      def setup(hexes:, tiles:, free: false, discount: 0)
         @hexes = hexes
         @tiles = tiles
         @free = free
+        @discount = discount
```

Comment:
it will be safer if we do discount: nil
@discount ||= 0

Reply:
Fixed